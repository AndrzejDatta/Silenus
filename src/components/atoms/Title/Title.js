import React from 'react';
import PropTypes from 'prop-types';
import { StyledTitle } from 'components/atoms/Title/Title.styles';
const Title = ({ title, pageType }) => {
  return <StyledTitle pageType={pageType}>{title}</StyledTitle>;
};

Title.propTypes = {
  /**text in Title atom*/
  title: PropTypes.string.isRequired,
  pageType: PropTypes.string.isRequired,
};

export default Title;
