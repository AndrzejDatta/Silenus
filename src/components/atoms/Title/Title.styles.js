import styled from "styled-components";
import { theme } from "assets/styles/theme";
export const StyledTitle = styled.h1`
  ${({ pageType }) => {
    if (pageType === "loginPage") {
      return `
        font-size: ${theme.font.size.heading};
        color: white;
        `;
    } else if (pageType === "startPage") {
      return `
        font-size: ${theme.font.size.heading};
        color: ${theme.color.fontGreen};
        `;
    }
  }}
  font-family: ${({ theme }) => theme.font.family.mitr};
  font-weight: 275;
`;
