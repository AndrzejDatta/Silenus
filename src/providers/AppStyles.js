import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
