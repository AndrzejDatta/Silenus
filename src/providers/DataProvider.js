import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
