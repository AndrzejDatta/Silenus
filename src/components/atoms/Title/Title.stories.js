import React from "react";
import Title from "components/atoms/Title/Title";
import { storiesName } from "helpers/storiesName";

export default {
  title: `${storiesName.atoms}Title`,
  component: Title,
};
const Template = (args) => <Title {...args} />;

export const Default = Template.bind({});

Default.args = {
  color: "black",
  title: "Title",
  size: "startPage",
};
