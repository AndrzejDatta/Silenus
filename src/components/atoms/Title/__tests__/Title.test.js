import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen, rerender } from "@testing-library/react";
import Title from "../Title";

describe("test Title component", () => {
  it("test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <Title />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
    expect(tree).toHaveStyleRule("color", "#3AA688");
  });
  if (
    ("Test Title component in loginPage",
    () => {
      render(
        <AppStyles>
          <Title pageType="loginPage" title="Silenus" />
        </AppStyles>
      );
      expect(screen.getByText(/Sielnus/i)).toHaveStyleRule("color", "white");
    })
  );
  if (
    ("Test Title component in startPage",
    () => {
      render(
        <AppStyles>
          <Title pageType="startPage" title="Silenus" />
        </AppStyles>
      );
      expect(screen.getByText(/Sielnus/i)).toHaveStyleRule("color", "#3AA688");
    })
  );
});
