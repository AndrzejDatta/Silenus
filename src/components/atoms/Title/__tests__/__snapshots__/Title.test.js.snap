// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Title component test default style 1`] = `
.c0 {
  font-size: 40px;
  color: #3AA688;
  font-family: "Mitr",sans-serif;
  font-weight: 275;
}

<h1
  className="c0"
>
  Title
</h1>
`;
